openedx/core/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -231,6 +231,36 @@ def test_jwt_access_token(self, client_attr):
             should_be_restricted=False,
         )
 
+    @ddt.data('dop_app', 'dot_app')
+    def test_jwt_access_token_from_header(self, client_attr):
+        client = getattr(self, client_attr)
+        response = self._post_request(self.user, client, HTTP_X_TOKEN_TYPE='jwt')
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content.decode('utf-8'))
+        self.assertIn('expires_in', data)
+        self.assertEqual(data['token_type'], 'JWT')
+        self.assert_valid_jwt_access_token(
+            data['access_token'],
+            self.user,
+            data['scope'].split(' '),
+            should_be_restricted=False,
+        )
+        
+    @ddt.data('dop_app', 'dot_app')
+    def test_jwt_access_token_from_parameter_not_header(self, client_attr):
+        client = getattr(self, client_attr)
+        response = self._post_request(self.user, client, HTTP_X_TOKEN_TYPE='invalid', token_type='jwt')
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content.decode('utf-8'))
+        self.assertIn('expires_in', data)
+        self.assertEqual(data['token_type'], 'JWT')
+        self.assert_valid_jwt_access_token(
+            data['access_token'],
+            self.user,
+            data['scope'].split(' '),
+            should_be_restricted=False,
+        )
```

Comment:
To keep our tests maintainable, we'll want to DRY these calls by creating a common helper function that these 3 tests use.  (Lines 253-262 are the same in each of these 3 tests.)